slycot/exceptions.py

```
@@ -0,0 +1,228 @@
+    in `checkvars`.
+
+    For negative info, the argument as indicated in arg_list was erroneous and
+    a generic SlycotParameterError is raised if matching infospec was defined.
+
+    Example
+    -------
+    >>> def fun(info):
+    ...     '''Example function
+    ...
+    ...     Raises
+    ...     ------
+    ...     SlycotArithmeticError : e
+    ...         :e.info = 1: Info is 1
+    ...         :e.info > 1 and e.info < n:
+    ...             Info is {e.info}, which is between 1 and {n}
+    ...         :n <= e.info < m:
+    ...             {e.info} is in [{n}, {m:10.2g})!
+    ...
+    ...     Warns
+    ...     -----
+    ...     SlycotResultWarning : e
+    ...         :e.info >= 120: {e.info} is too large
+    ...     '''
+    ...     n, m = 4, 120.
+    ...     raise_if_slycot_error(info,
+    ...                           arg_list=["a", "b", "c"],
+    ...                           docstring=fun.__doc__,
+    ...                           checkvars=locals())
+    ...
+    >>> fun(0)
+    >>> fun(-1)
+    SlycotParameterError:
+    The following argument had an illegal value: a
+    >>> fun(1)
+    SlycotArithmeticError:
+    Info is 1
+    >>> fun(2)
+    SlycotArithmeticError:
+    Info is 2, which is between 1 and 4
+    >>> fun(4)
+    SlycotArithmeticError:
+    4 is in [4,    1.2e+02)!
+    >>> fun(120)
+    SlycotResultWarning:
+    120 is too large
+    """
+    if docstring:
+        checkvars['e'] = SlycotError("", info)
+
+        exception, message = _parse_docsection("Raises", docstring, checkvars)
+        if exception and message:
+            fmessage = '\n' + message.format(**checkvars).strip()
+            raise globals()[exception](fmessage, info)
+
+        warning, message = _parse_docsection("Warns", docstring, checkvars)
+        if warning and message:
+            fmessage = '\n' + message.format(**checkvars).strip()
+            warn(globals()[warning](fmessage, info))
+
```

and if we don't find a warning or message, and info > 0, we should return a catch-all exception, with just the info number. Better safe than sorry.

---

I agree, that now with the possibility to mention warnings, a catchall can be implemented.